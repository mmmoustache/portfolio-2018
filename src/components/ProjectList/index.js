import React from 'react';
import PropTypes from 'prop-types';
import Card from "../ProjectCard";
import TrackVisibility from 'react-on-screen';

const offset = -300;

const ProjectList = ({ items }) => {
  return (
    <ul id="work">
      { 
        items && items.map((e, i) => 
          <li key={e.id}>
            <TrackVisibility offset={offset} partialVisibility once>
              {({ isVisible }) => 
                <Card
                  title={e.title}
                  description={e.description}
                  image_url={e.image_url}
                  technologies={e.technologies}
                  url={e.url}
                  repo={e.repo}
                  isVisible={isVisible}
                  isOdd={i % 2 === 0}
                />
              }   
            </TrackVisibility>            
          </li>
        )
      }
    </ul>
  )
};

ProjectList.propTypes = {
  items: PropTypes.array,
};

ProjectList.defaultProps = {
  items: undefined,
};

export default ProjectList;
