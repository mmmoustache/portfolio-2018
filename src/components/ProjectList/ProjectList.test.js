import React from 'react';
import renderer from 'react-test-renderer';
import ProjectList from "../ProjectList";

describe('Component: ProjectList', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <ProjectList items={[]} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
