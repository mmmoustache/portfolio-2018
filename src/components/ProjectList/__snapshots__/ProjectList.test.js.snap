// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProjectList Should render successfully 1`] = `
<ul
  id="work"
/>
`;
